src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -241,12 +238,13 @@ def on_left_down(self, evt):
             self._is_new_shape = True
         # New or previously created shape
         else:
-            self._selected_shape = self._get_shape(evt)
+            self._selected_shape = self._get_shape(evt.Position)
             if self._selected_shape is None:
                 self._selected_shape = self._create_new_shape()
                 self._is_new_shape = True
-            self._selected_shape.active.value = True
-            self._selected_shape.on_left_down(evt)
+            if hasattr(self._selected_shape, "on_left_down"):
+                self._selected_shape.on_left_down(evt)
```

Comment:
Please make all shapes have a `on_left_down` method, so it's always safe to call it. (It might do just nothing). It'll be faster than checking first. Same thing with the other methods below.